modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -295,11 +295,6 @@
   {:arglists '([clause])}
   mbql.u/dispatch-by-clause-name-or-class)
 
-(def ^:private ^:dynamic *top-level-filter?*
-  "Whether we are compiling a top-level filter clause. This means we can generate somewhat simpler `$match` clauses that
-  don't need to use `$expr` (see below)."
-  true)
-
```

Comment:
I think this was needed for MongoDB 3.0 or something (some ancient unsupported version) but things totally work fine now without it.